src/FirmwarePlugin/APM/APMSensorParams.qml

```diff
@@ -17,15 +17,23 @@ Item {
 
     property Fact _noFact: Fact { }
 
-    property Fact compassPrimaryFact:               factPanelController.getParameterFact(-1, "COMPASS_PRIMARY")
-    property bool compass1Primary:                  compassPrimaryFact.rawValue == 0
-    property bool compass2Primary:                  compassPrimaryFact.rawValue == 1
-    property bool compass3Primary:                  compassPrimaryFact.rawValue == 2
+    property bool compassPrimaryFactAvailable:      factPanelController.parameterExists(-1, "COMPASS_PRIMARY")
+    property Fact compassPrimaryFact:               compassPrimaryFactAvailable ? factPanelController.getParameterFact(-1, "COMPASS_PRIMARY") : _noFact
+    property bool compass1Primary:                  compassPrimaryFactAvailable ? compassPrimaryFact.rawValue == 0 : false
+    property bool compass2Primary:                  compassPrimaryFactAvailable ? compassPrimaryFact.rawValue == 1 : false
+    property bool compass3Primary:                  compassPrimaryFactAvailable ? compassPrimaryFact.rawValue == 2 : false
     property var  rgCompassPrimary:                 [ compass1Primary, compass2Primary, compass3Primary ]
 
     property Fact compass1Id:                       factPanelController.getParameterFact(-1, "COMPASS_DEV_ID")
     property Fact compass2Id:                       factPanelController.getParameterFact(-1, "COMPASS_DEV_ID2")
     property Fact compass3Id:                       factPanelController.getParameterFact(-1, "COMPASS_DEV_ID3")
+    property var  rgCompassId:                      [ compass1Id, compass2Id, compass3Id ]
+
+    property bool compassPrioFactsAvailable:        factPanelController.parameterExists(-1, "COMPASS_PRIO1_ID")
```

Comment:
Does this really guarantee that `COMPASS_PRIO2_ID` and `COMPASS_PRIO3_ID` are available as well?